PredixMobileUIKit/PredixBarChartView.swift

```diff
@@ -0,0 +1,330 @@
+            var dataEntries: [BarChartDataEntry] = []
+            for i in 0 ..< bar.values.count {
+                let dataEntry = BarChartDataEntry(x: Double(i), y: bar.values[i])
+                dataEntries.append(dataEntry)
+            }
+            let chartDataSet = BarChartDataSet(values: dataEntries, label: bar.label)
+            if !bar.colors.isEmpty {
+                chartDataSet.colors = bar.colors
+            }
+            dataSets.append(chartDataSet)
+        }
+        return dataSets
+    }
+
+    /// Stacks all Bar DataSets in the Chart
+    internal func stackBarsOnChart() {
+        if !bars.isEmpty {
+
+            let groupCount = xAxisValues.count
+            let xaxis = xAxis
+            xaxis.valueFormatter = IndexAxisValueFormatter(values: xAxisValues)
+
+            let stackBarsDataSets = createDataSets(bars: bars)
+            let stackBarsCharData = BarChartData(dataSets: stackBarsDataSets)
+
+            let stackBarsStatingNumber = -0.5
+            xaxis.centerAxisLabelsEnabled = false
+            xaxis.axisMinimum = stackBarsStatingNumber
+            xaxis.axisMaximum = stackBarsStatingNumber + Double(groupCount)
+            data = stackBarsCharData
+        }
+    }
+
+    /// Groups all Bar DataSets in the Chart
+    internal func groupBarsOnChart() {
+        if !bars.isEmpty {
+
+            let groupCount = xAxisValues.count
+            let xaxis = xAxis
+            xaxis.valueFormatter = IndexAxisValueFormatter(values: xAxisValues)
+
+            let dataSets = createDataSets(bars: bars)
+            let groupBarsChartData = BarChartData(dataSets: dataSets)
+            let barCount = groupBarsChartData.dataSets.count
+            let barSpace = 0.05
+            let groupBarsStatingNumber = 0.0
+
+            /*
+             The equation to find the interval per "group" is:
+             (groupSpace * barSpace) * n + groupSpace = 1
+             Therefore by finding groupSpace we get:
+             groupSpace = 1 - numberOfBars * BarSpace / numberOfBars +1
+             */
+            let groupSpace = (1.0 - Double(barCount) * barSpace) / (Double(barCount) + 1.0)
+            let barWidth = groupSpace
+
+            groupBarsChartData.barWidth = barWidth
+            xaxis.centerAxisLabelsEnabled = true
+
+            xaxis.axisMinimum = groupBarsStatingNumber
+            let individualGroupSpace = groupBarsChartData.groupWidth(groupSpace: groupSpace, barSpace: barSpace)
+            xaxis.axisMaximum = groupBarsStatingNumber + individualGroupSpace * Double(groupCount)
+            groupBarsChartData.groupBars(fromX: Double(groupBarsStatingNumber), groupSpace: groupSpace, barSpace: barSpace)
+            data = groupBarsChartData
+        }
+    }
+
+    // MARK: PredixBarChartView  Handling Option Logic
+
+    /// Handle the options provided.
+    /// - parameter option: choose one of the option from the Option class.
+    public func handleOption(_ option: Option) {
```

Comment:
Can the Option enum have multiple settings? If not, this method seems awkward, What If I want to toggleValues and animateX at the same time? The structure of the method suggests all these options are related, and wouldn't be combined, but the names of the Options enumeration suggests otherwise.